File: event-bus-rabbit/src/main/java/pegasus/eventbus/topology/event/RegisterClient.java
package pegasus.eventbus.topology.event;

public class RegisterClient extends Registration {

    private String version;

    //@todo - needed for gson in osgi
    public RegisterClient() {
        
    }
    
    public RegisterClient(String clientName, String version) {
        super(clientName);
        this.version = version;
    }

    public String getVersion() {
        return version;
    }
    
    public void setVersion(String version) {
        this.version = version;
    }

}

File: event-bus-rabbit/src/main/dotnet/topology/events/RegisterClient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace pegasus.eventbus.topology.events
{
    public class RegisterClient : Registration
    {
        public string Version;


        public RegisterClient() { }
        public RegisterClient(string clientName, string version) 
            : base(clientName)
        {
            this.Version = version;
        }
    }
}
